mbrl/util/huggingface.py

```diff
@@ -0,0 +1,351 @@
+...
+```
+"""
+
+    return model_card, metadata
+
+
+def _save_model_card(
+    local_path: Path, generated_model_card: str, metadata: Dict[str, Any]
+):
+    """Saves a model card for the repository.
+    :param local_path: repository directory
+    :param generated_model_card: model card generated by _generate_model_card()
+    :param metadata: metadata
+    """
+    readme_path = local_path / "README.md"
+    readme = ""
+    if readme_path.exists():
+        with readme_path.open("r", encoding="utf8") as f:
+            readme = f.read()
+    else:
+        readme = generated_model_card
+
+    with readme_path.open("w", encoding="utf-8") as f:
+        f.write(readme)
+
+    # Save our metrics to Readme metadata
+    metadata_save(readme_path, metadata)
+
+
+def _add_logdir(local_path: Path, logdir: Path):
+    """Adds a logdir to the repository.
+    :param local_path: repository directory
+    :param logdir: logdir directory
+    """
+    if logdir.exists() and logdir.is_dir():
+        # Add the logdir to the repository under new dir called logs
+        repo_logdir = local_path / "logs"
+
+        # Delete current logs if they exist
+        if repo_logdir.exists():
+            shutil.rmtree(repo_logdir)
+
+        # Copy logdir into repo logdir
+        shutil.copytree(logdir, repo_logdir)
+
+
+def package_to_hub(
+    model: Model,
+    model_name: str,
+    model_architecture: str,
+    env_id: str,
+    eval_env: gym.Env,
+    repo_id: str,
+    commit_message: str,
+    is_deterministic: bool = True,
+    n_eval_episodes=10,
+    token: Optional[str] = None,
+    video_length=1000,
+    logs=None,
+    mean_reward=None,
+    std_reward=None,
+):
+    """
+    Evaluate, Generate a video and Upload a model to Hugging Face Hub.
+    This method does the complete pipeline:
+    - It evaluates the model
+    - It generates the model card
+    - It generates a replay video of the agent
+    - It pushes everything to the hub
+    :param model: trained model
+    :param model_name: name of the model zip file
+    :param model_architecture: name of the architecture of your model
+        (DQN, PPO, A2C, SAC...)
+    :param env_id: name of the environment
+    :param eval_env: environment used to evaluate the agent
+    :param repo_id: id of the model repository from the Hugging Face Hub
+    :param commit_message: commit message
+    :param is_deterministic: use deterministic or stochastic actions (by default: True)
+    :param n_eval_episodes: number of evaluation episodes (by default: 10)
+    :param video_length: length of the video (in timesteps)
+    :param logs: directory on local machine of tensorboard logs you'd like to upload
+    """
+
+    msg = (
+        "This function will save your agent, "
+        "create a model card and push everything to the hub. "
+        "This is a work in progress: if you encounter a bug, please open an issue."
+    )
+    print(msg)
+
+    repo_url = HfApi().create_repo(
```

Comment:
(nit) for consistency with `upload_folder` calls, I would import `create_repo` directly from `huggingface_hub`.
If not called with any parameter, using `HfApi().` is not needed.